src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php

```diff
@@ -60,12 +61,21 @@ public function createLoginLink(UserInterface $user): LoginLinkDetails
             'hash' => $this->computeSignatureHash($user, $expires),
         ];
 
+        if ($request) {
+            $currentRequestContext = $this->urlGenerator->getContext();
+            $this->urlGenerator->setContext((new RequestContext())->fromRequest($request));
+        }
+
         $url = $this->urlGenerator->generate(
             $this->options['route_name'],
             $parameters,
             UrlGeneratorInterface::ABSOLUTE_URL
         );
 
+        if ($currentRequestContext ?? false) {
```

Comment:
this should be `if ($request)` instead
also, what about nesting this in a `finally` block?